eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs

```diff
@@ -835,6 +847,7 @@ instance Era era => DecCBOR (Annotator (AlonzoScript era)) where
       decodeScript 0 = Ann (SumD TimelockScript) <*! From
       decodeScript 1 = Ann (SumD $ PlutusScript PlutusV1) <*! Ann From
       decodeScript 2 = Ann (SumD $ PlutusScript PlutusV2) <*! Ann From
+      decodeScript 3 = Ann (SumD $ PlutusScript PlutusV3) <*! Ann From
```

Comment:
We need to forbid deserialization of PlutusV3 before `natVersion @9`  here too. This would take care of the inline script in the TxOut too.